pkg/doc/vc/crypto/sdjwt.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package crypto
+
+import (
+	"crypto"
+	"fmt"
+	"strings"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/sdjwt/common"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/sdjwt/issuer"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/verifiable"
+
+	"github.com/trustbloc/vcs/pkg/doc/vc"
+)
+
+// createSDJWT creates SD-JWT based on credential.
+//
+//	 Next steps are taken:
+//		1. Take the credential subject in map[string]interface{} format and remove "id" key from it
+//			(in order to reduce the amount of digests).
+//		2. Create issuer.SelectiveDisclosureJWT based on credential subject.
+//		3. OVERRIDE original credential.Subject with calculated digests from prev step.
+//		4. Create JWS based on credential.
+//		5. Append Disclosures from issuer.SelectiveDisclosureJWT to JWS from step 4.
+//	 Note, that current implementation of SD-JWT does not support credential subject in a string format.
+//	 Note, that Holder binding is not supported yet
+//	  https://www.ietf.org/archive/id/draft-ietf-oauth-selective-disclosure-jwt-02.html#name-holder-binding
+func (c *Crypto) createSDJWT(
```

Comment:
I would suggest - having new function in aries which returns you digests & disclosures as return arguments and you build SD-JWT from it.

 Looks like here we are,
- signing once in aries
- manipulating response & build new VC
- later signing it again in VCS

Ideally, you just build an unsigned SD-JWT VC & later sign it using VCS signer interface.

Reply:
You can create unsigned SD-JWT as per https://github.com/hyperledger/aries-framework-go/pull/3480 and use payload(digests) and disclosures from it.

Reply:
@sudeshrshetty @sandrask updatd. Pls see latest updates.